packages/replay-next/src/hooks/useTooltip.tsx

```diff
@@ -8,6 +17,129 @@ type TooltipPosition = "above" | "below" | "left-of" | "right-of";
 export const MARGIN_LARGE = 8;
 export const MARGIN_SMALL = 4;
 
+// A copy-pasted version of the other `useTooltip` hook,
+// but exposing `setShowTooltip` for control from the parent component
+// instead of controlling display itself on hover.
+export function usePositionedTooltip({
+  className,
+  containerRef,
+  position,
+  tooltip,
+  targetRef,
+}: {
+  className?: string;
+  containerRef?: RefObject<HTMLElement>;
+  position: TooltipPosition;
+  targetRef: RefObject<HTMLElement>;
+  tooltip: ReactNode;
+}): {
+  setShowTooltip: Dispatch<SetStateAction<boolean>>;
```

Comment:
This feels awkward to me. I think maybe a more straight forward approach would be to update `useTooltip` to accept an optional flag `forceVisible` or something that would override the internal state when present.

```ts
export default function useTooltip({
  className,
  containerRef,
  delay = 0,
  forceVisible = false,
  position,
  tooltip,
}: {
  className?: string;
  containerRef?: RefObject<HTMLElement>;
  delay?: number;
  forceVisible?: boolean;
  position?: TooltipPosition;
  tooltip: ReactNode;
}): {
  onMouseEnter: (event: MouseEvent) => void;
  onMouseLeave: (event: MouseEvent) => void;
  onMouseMove: (event: MouseEvent) => void;
  tooltip: ReactNode;
} {
```

Then I don't think we'd need to fork at all?

Reply:
Okay, I'll give that a shot.

Reply:
@bvaughn : ah, I remember why I really ended up copy-pasting this.

`useTooltip` _expects_ that you're using its `onMouseEnter/onMouseLeave` callbacks in order to get a reference to the target element.  It doesn't have a way to pass that in.

It was easier to fork the entire hook and swap that out to be an argument instead.

Reply:
k. Fork it then, but I think it should be in its own file. Mixing named and default exports like this seems confusing to me.

Seems like we could share the _position_ logic between the two though.

Reply:
I guess the _other_ options would be to change `useTooltip` to accept a `targetRef` prop but then that makes the regular "tooltip" use cases a little more verbose so maybe it's not worth it.

I dunno. End of a long week and my brain is feeling friend tbh.

Reply:
Okay. Personally I've shifted to not being a fan of default exports in the first place :)  But that's a separate discussion.

And yeah, extracting the positioning logic was what I started with, but that's also its own rabbit hole and I didn't want to deal with that for now.

Reply:
> Personally I've shifted to not being a fan of default exports in the first place

Yeah. I think I've been coming around to that position myself, although default exports do have a nice benefit of speeding up file navigation for me in VS Code (e.g. Cmd+P -> type name of function).